packages/interactivity/docs/2-api-reference.md

```diff
@@ -0,0 +1,617 @@
+    - [`wp-style`](#wp-style) ![](https://img.shields.io/badge/ATTRIBUTES-afd2e3.svg)
+    - [`wp-text`](#wp-text) ![](https://img.shields.io/badge/CONTENT-afd2e3.svg)
+    - [`wp-on`](#wp-on) ![](https://img.shields.io/badge/EVENT_HANDLERS-afd2e3.svg)
+    - [`wp-effect`](#wp-effect) ![](https://img.shields.io/badge/SIDE_EFFECTS-afd2e3.svg)
+    - [`wp-init`](#wp-init) ![](https://img.shields.io/badge/SIDE_EFFECTS-afd2e3.svg)
+  - [Values of directives are references to properties](#values-of-directives-are-references-to-properties)
+- [The store](#the-store)
+  - [Elements of the store](#elements-of-the-store)
+    - [State](#state)
+    - [Actions](#actions)
+    - [Effects](#effects)
+    - [Selectors](#selectors)
+  - [Objects passed to callbacks](#objects-passed-to-callbacks)
+  - [Setting the store](#setting-the-store)
+    - [On the client side](#on-the-client-side)
+    - [On the server side](#on-the-server-side)
+
+
+
+## The directives
+
+Directives are custom attributes that are added to the markup of your block to add behaviour to its DOM elements. This can be done in the `render.php` file (for dynamic blocks) or the `save.js` file (for static blocks).
+
+Interactivity API directives use the `data-` prefix.
+
+_Example of directives used in the HTML markup_
+
+```html
+<div
+  data-wp-context='{ "myNamespace" : { "isOpen": false } }'
+  data-wp-effect="effects.myNamespace.logIsOpen"
+>
+  <button
+    data-wp-on--click="actions.myNamespace.toggle"
+    data-wp-bind--aria-expanded="context.myNamespace.isOpen"
+    aria-controls="p-1"
+  >
+    Toggle
+  </button>
+ 
+  <p id="p-1" data-bind--hidden="!context.myNamespace.isOpen">
+    This element is now visible!
+  </p>
+</div>
+```
+
+> **Note**
+> The use of Namespaces to define the context, state or any other elements of the store is highly recommended to avoid possible collision with other elements with the same name. In the following examples we have not used namespaces for the sake of simplicity.
+
+Directives can also be injected dynamically using the [HTML Tag Processor](https://make.wordpress.org/core/2023/03/07/introducing-the-html-api-in-wordpress-6-2).
+
+### List of Directives
+
+With directives we can manage directly in the DOM behavior related to things such as side effects, state, event handlers, attributes or content.
+
+#### `wp-interactive` 
+
+The `wp-interactive` "activates" hydration for the DOM element and its children through the Interactivity API (directives and store). 
+
+> **Note**
+> The use of `wp-interactive` is a requirement for the Interactivity API "engine" to work. In the following examples the `wp-interactive` has not been added for the sake of simplicity.
+
+
+#### `wp-context` 
+
+It provides **local** state available to a specific HTML node and its children.
+
+The `wp-context` directive accepts a stringified JSON as value.
+
+_Example of `wp-context` directive_ 
+```php
+//render.php
```

Comment:
Because of the line `<div data-wp-context='{ {"post": { "id": <?php echo $post->ID; ?> } } ' >` wouldn't this code just be possible to be ran from `render.php`?

Reply:
Ohhh, you're right 😅